packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -139,19 +169,47 @@ export const buildPropertyExpressionFromExplorerTreeNodeData = (
     multiplicityOne,
   );
   propertyExpression.func = guaranteeNonNullable(node.property);
-  let currentExpression = propertyExpression;
+  let currentExpression: AbstractPropertyExpression | SimpleFunctionExpression =
+    propertyExpression;
   let parentNode = treeData.nodes.get(node.parentId);
-  while (parentNode instanceof QueryBuilderExplorerTreePropertyNodeData) {
-    const parentPropertyExpression = new AbstractPropertyExpression(
-      '',
-      multiplicityOne,
-    );
-    parentPropertyExpression.func = guaranteeNonNullable(parentNode.property);
+  let currentNode: QueryBuilderExplorerTreeNodeData = node;
+  while (
+    parentNode instanceof QueryBuilderExplorerTreePropertyNodeData ||
+    parentNode instanceof QueryBuilderExplorerTreeSubTypeNodeData
+  ) {
+    let parentPropertyExpression;
+    if (parentNode instanceof QueryBuilderExplorerTreeSubTypeNodeData) {
+      parentPropertyExpression = new SimpleFunctionExpression(
+        'subType',
+        multiplicityOne,
+      );
+    } else {
+      parentPropertyExpression = new AbstractPropertyExpression(
+        '',
+        multiplicityOne,
+      );
+      parentPropertyExpression.func = guaranteeNonNullable(parentNode.property);
+    }
     currentExpression.parametersValues.push(parentPropertyExpression);
+    if (currentExpression instanceof SimpleFunctionExpression) {
```

Comment:
we should be probably checking the function name as well, not just the instance. we should assert its a subtype function.